src/utils/table-list-utils.ts

```diff
@@ -0,0 +1,34 @@
+import { SortByDirection, ThProps } from '@patternfly/react-table';
+import { HeaderFunc } from '~/shared/components/table/Table';
+
+export type ColumnConfig = {
+  title: string;
+  className: string;
+  sortable?: boolean;
+  style?: React.CSSProperties;
+};
+
+export const createTableHeaders = (
+  columnConfig: ColumnConfig[],
+): ((
+  activeIndex: number,
+  activeDirection: SortByDirection,
+  onSort: ThProps['sort']['onSort'],
+) => HeaderFunc) => {
+  return (activeIndex, activeDirection, onSort) => () => {
+    const getSortParams = (columnIndex: number) => ({
+      columnIndex,
+      sortBy: { index: activeIndex, direction: activeDirection },
+      onSort,
+    });
+
```

Comment:
These are specific to Table component. Can we move these to shared folder with the Table component itself.

Reply:
updated! :)